frontend/src/concepts/pipelines/content/compareRuns/metricsSection/confusionMatrix/ConfusionMatrixCompare.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+import {
+  Bullseye,
+  Divider,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateHeader,
+  EmptyStateVariant,
+  Flex,
+  FlexItem,
+  Spinner,
+} from '@patternfly/react-core';
+import {
+  ConfusionMatrixConfig,
+  buildConfusionMatrixConfig,
+} from '~/concepts/pipelines/content/artifacts/charts/ConfusionMatrix';
+import { PipelineRunKFv2 } from '~/concepts/pipelines/kfTypes';
+import { RunArtifact } from '~/concepts/pipelines/apiHooks/mlmd/types';
+
+import { FullArtifactPath } from '~/concepts/pipelines/content/compareRuns/metricsSection/types';
+import {
+  getFullArtifactPaths,
+  getFullArtifactPathLabel,
+} from '~/concepts/pipelines/content/compareRuns/metricsSection/utils';
+import { CompareRunsEmptyState } from '~/concepts/pipelines/content/compareRuns/CompareRunsEmptyState';
+import { ConfusionMatrixSelect } from './ConfusionMatrixSelect';
+import { ConfusionMatrixConfigAndTitle } from './types';
+import { isConfusionMatrix } from './utils';
+
+type ConfusionMatrixCompareProps = {
+  runArtifacts?: RunArtifact[];
+  isLoaded: boolean;
+};
+
+const ConfusionMatrixCompare: React.FC<ConfusionMatrixCompareProps> = ({
+  runArtifacts,
+  isLoaded,
+}) => {
+  const [expandedGraph, setExpandedGraph] = React.useState<
+    ConfusionMatrixConfigAndTitle | undefined
+  >(undefined);
+
+  const fullArtifactPaths: FullArtifactPath[] = React.useMemo(() => {
+    if (!runArtifacts) {
+      return [];
+    }
+
+    return getFullArtifactPaths(runArtifacts);
+  }, [runArtifacts]);
+
+  const { configMap, runMap } = React.useMemo(() => {
+    const configMapBuilder: Record<string, { title: string; config: ConfusionMatrixConfig }[]> = {};
+    const runMapBuilder: Record<string, PipelineRunKFv2> = {};
+
+    fullArtifactPaths
+      .map((fullPath) => {
+        const customProperties = fullPath.linkedArtifact.artifact.getCustomPropertiesMap();
+        return {
+          run: fullPath.run,
+          title: getFullArtifactPathLabel(fullPath),
+          data: customProperties.get('confusionMatrix')?.getStructValue()?.toJavaScript(),
+        };
+      })
+      .filter((confidenceMetrics) => !!confidenceMetrics.data)
```

Comment:
At least, this map & filter can be simplified with 1 reduce, where probably the forEach could be combined with that reduce as well.